src/hooks/v2/NftRewards.ts

```diff
@@ -0,0 +1,33 @@
+import axios from 'axios'
+import { NftRewardTier } from 'models/v2/nftRewardTier'
+import { useQuery, UseQueryResult } from 'react-query'
+import { ipfsCidUrl } from 'utils/ipfs'
+
+async function getRewardTierOfCid(cid: string) {
+  const url = ipfsCidUrl(cid)
+  const response = await axios.get(url)
+  return response.data
+}
+
+// Retreives each NftRewardTier from IPFS given an array of CIDs (IpfsHashes)
+// Returns an array of NftRewardTiers
+export default function useNftRewards(
+  CIDs: string[] | undefined,
+): UseQueryResult<NftRewardTier[]> {
+  return useQuery('nft-rewards', async () => {
+    if (!CIDs?.length) {
+      return
+    }
+
+    console.info('>>> Retreiving reward tiers from IPFS using CIDs: ', CIDs)
+    const getRewardTiers = async () => {
+      return Promise.all(CIDs.map(cid => getRewardTierOfCid(cid)))
```

Comment:
**question** just checkin: CIDS.length is max 3 right?

If its any bigger, this line may be a problem. But for max 3 tiers, this is chill.

Reply:
Actually, it's not enforced yet. Will do

Reply:
Done